soroban-env-host/src/auth.rs

```diff
@@ -736,7 +758,12 @@ impl AuthorizationTracker {
                 Some(curr_invocation.sub_invocations.len() - 1);
         } else {
             // This would be a bug
-            return Err(host.err_general("unexpected missing authorized invocation"));
+            return Err(host.err(
+                ScErrorType::Auth,
+                ScErrorCode::MissingValue,
```

Comment:
nit: this should be `InternalError` probably (given that this is an 'implementation bug' state).

The same actually applies to all the errors below besides 'nonce is incorrect' one. Basically anything starting with `unexpected` is an internal error and shouldn't ever happen (unless we have bugs).

Reply:
ok I'll fix these

Reply:
fixed (all except the ones about nonces in the ledger having the wrong type -- that seems like a user error)

Reply:
These would actually be internal as well as the host manages nonces exclusively currently. But nonces also will hopefully change the implementation after temp storage introduction, so also feel free to skip these for now, it doesn't really matter that much.